tests/unit/test_help.py

```diff
@@ -92,3 +95,227 @@ def test_shlex_split_for_pager_var(self):
         os.environ['PAGER'] = pager_cmd
         self.assertEqual(self.renderer.get_pager_cmdline(),
                          ['/bin/sh', '-c', "col -bx | vim -c 'set ft=man' -"])
+
+
+class TestHelpCommandBase(unittest.TestCase):
+    def setUp(self):
+        self.session = mock.Mock()
+        self.file_creator = FileCreator()
+
+    def tearDown(self):
+        self.file_creator.remove_all()
+
+
+class TestProviderHelpCommand(TestHelpCommandBase):
+    def setUp(self):
+        super(TestProviderHelpCommand, self).setUp()
+        self.session.provider = None
+        self.command_table = {}
+        self.arg_table = {}
+        self.description = None
+        self.synopsis = None
+        self.usage = None
+
+        # Create a temporary index file for ``aws help [command]`` to use.
+        self.tags_dict = {
+            'topic-name-1': {},
+            'topic-name-2': {}
+        }
+        json_index = self.file_creator.create_file('index.json', '')
+        with open(json_index, 'w') as f:
+            json.dump(self.tags_dict, f, indent=4, sort_keys=True)
+        self.json_patch = mock.patch(
+            'awscli.topictags.TopicTagDB.index_file', json_index)
+        self.json_patch.start()
+
+        self.cmd = ProviderHelpCommand(self.session, self.command_table,
+                                       self.arg_table, self.description,
+                                       self.synopsis, self.usage)
+
+    def tearDown(self):
+        self.json_patch.stop()
+        super(TestProviderHelpCommand, self).tearDown()
+
+    def test_topic_table(self):
+        topic_table = self.cmd.topic_table
+
+        self.assertEqual(len(topic_table), 3)
+
+        # Ensure there is a topics command
+        self.assertIn('topics', topic_table)
+        self.assertIsInstance(topic_table['topics'], TopicListerCommand)
+
+        # Ensure the topics are there as well
+        self.assertIn('topic-name-1', topic_table)
+        self.assertIsInstance(topic_table['topic-name-1'], TopicHelpCommand)
+        self.assertEqual(topic_table['topic-name-1'].name, 'topic-name-1')
+
+        self.assertIn('topic-name-2', topic_table)
+        self.assertIsInstance(topic_table['topic-name-2'], TopicHelpCommand)
+        self.assertEqual(topic_table['topic-name-2'].name, 'topic-name-2')
+
+    def test_topics_call(self):
+        with mock.patch('awscli.help.TopicListerCommand.__call__') \
+                as mock_call:
+            self.cmd(['topics'], None)
+            mock_call.assert_called()
+
+    def test_topic_call(self):
+        with mock.patch('awscli.help.TopicHelpCommand.__call__') as mock_call:
+            self.cmd(['topic-name-1'], None)
+            mock_call.assert_called()
+
+    def test_regular_call(self):
+        with mock.patch('awscli.help.TopicListerCommand.__call__') \
+                as topics_call:
+            with mock.patch('awscli.help.TopicHelpCommand.__call__') \
```

Comment:
Decorators should make this a bit less painful. For example:

```python
@mock.patch('awscli.help.TopicListerCommand.__call__')
@mock.patch('awscli.help.TopicHelpCommand.__call__')
@mock.patch('awscli.help.HelpCommand.__call__')
def test_regular_call(self, help_command, topic_help_command, topic_lister_command):
    self.cmd([], None)
    help_command.assert_called()
    self.assertFalse(topic_lister_command.called)
    self.assertFalse(topic_help_command.called)
```

Still not perfect, but it's a little cleaner IMO.